operator/pkg/ciliumidentity/reconciler.go

```diff
@@ -163,6 +162,13 @@ func (r *reconciler) reconcileCID(cidResourceKey resource.Key) error {
 			return r.createCID(cidName, cidKey)
 		} else {
 			r.desiredCIDState.Remove(cidName)
```

Comment:
these two methods should always be called together, so maybe `makeIDAvailable` should call `r.desiredCIDState.Remove(cidName)` internally?

Reply:
Done